### PR TITLE
Fix for https://github.com/composer/composer/issues/5989

### PR DESCRIPTION
**Pull request by theotonge**
This adds makes the init command enforce yes/no answers for boolean questions as requested in issue #5989

### CODE REVIEW

File: src/Composer/Question/StrictConfirmationQuestion.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    private $trueAnswerRegex;
+    private $falseAnswerRegex;
+
+    /**
+     * Constructor.s
+     *
+     * @param string $question        The question to ask to the user
+     * @param bool   $default         The default answer to return, true or false
+     * @param string $trueAnswerRegex A regex to match the "yes" answer
+     */
```
Comment (amanmahajan26) [inline src/Composer/Question/StrictConfirmationQuestion.php]: missing param for $falseAnswerRegex

Comment (theotonge) [inline src/Composer/Question/StrictConfirmationQuestion.php]: @amanmahajan26 Thank you, I have pushed a corrected version with this fixed

**Comment by Seldaek** [comment]
Thanks for the PR. However I think this change/fix should be done inside ConsoleIO::askConfirmation so that all askConfirmation calls become more strict, instead of just changing those in init command, which also means the init command doesn't use the IOInterface consistently anymore..

**Comment by Seldaek** [comment]
Also if you can rebase this on the 1.4 branch and change the pull request to target 1.4 that would be nice as it's a bugfix it can be applied to the next 1.4.x :)

**Comment by theotonge** [comment]
I have updated accordingly - but forgot to rerun tests :( Will fix failed tests and update again

**Comment by Seldaek** [comment]
Thanks :)

**Pull request merged**